proteus/LinearSolvers.py

```diff
@@ -809,38 +790,48 @@ def getFp(self,output_matrix=False):
         self.Fsys_rowptr = rowptr.copy()
         self.Fsys_colind = colind.copy()
         self.Fsys_nzval = nzval.copy()
+        self.Fsys_nzval.fill(0.0)
+
         L_sizes = self.L.getSizes()
         nr = L_sizes[0][0]
         nc = L_sizes[1][0]
-        self.Fsys =SparseMat(nr,nc,
-                             self.Fsys_nzval.shape[0],
-                             self.Fsys_nzval,
-                             self.Fsys_colind,
-                             self.Fsys_rowptr)
+
+        self.Fsys = SparseMat(nr,nc,
+                              self.Fsys_nzval.shape[0],
+                              self.Fsys_nzval,
+                              self.Fsys_colind,
+                              self.Fsys_rowptr)
         self.L.pde.q[('df',0,0)][...,0] = self.L.pde.q[('u',1)]
         self.L.pde.q[('df',0,0)][...,1] = self.L.pde.q[('u',2)]
-        import pdb
  #        self.L.pde.q[('df',0,0)][...,2] = self.L.pde.q[('u',3)]
         self.L.pde.getSpatialJacobian(self.Fsys)#notice, switched  to spatial
         self.Fsys_petsc4py = self.L.duplicate()
+
         F_csr_rep_local = self.Fsys.getSubMatCSRrepresentation(0,L_sizes[0][0])
         self.Fsys_petsc4py.setValuesLocalCSR(F_csr_rep_local[0],
                                              F_csr_rep_local[1],
                                              F_csr_rep_local[2],
                                              p4pyPETSc.InsertMode.INSERT_VALUES)
+
         self.Fsys_petsc4py.assemblyBegin()
         self.Fsys_petsc4py.assemblyEnd()
-        self.Cp = self.Fsys_petsc4py.getSubMatrix(self.linear_smoother.isp,self.linear_smoother.isp)
+
+        self.Cp = self.Fsys_petsc4py.getSubMatrix(self.linear_smoother.isp,
+                                                  self.linear_smoother.isp)
+
```

Comment:
`self.Cp` should be the pressure-advection term that needs to be added to the scaled pressure-lapacian to create the correct Fp operator.  Based on my simulations this does not appear to be creating the proper matrix.